custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/%s' % __short_version__
+        self.session.headers['Content-Type'] = 'application/json'
+        self.session.headers['Authorization'] = self.api_key
+
+    def terminate(self):
+        self.session.close()
+
+    def query(self, video):
+        series_name = (
+            video.alternative_series[0] if len(video.alternative_series) > 0 else str(video.series)
+        ).lower()
+
+        # Check if series_name ends with "Sn", if so strip chars as Jimaku only lists seasons by numbers alone
+        # If 'n' is 1, completely strip it as first seasons don't have a season number
+        if re.search(r's\d$', series_name):
+            if int(series_name[-1]) == 1:
+                series_name = re.sub(r's\d$', "", series_name)
+            else:
+                # As shows sometimes have the wrong season number in the title, we'll reassemble it
+                series_name = re.sub(r's\d$', str(video.season), series_name)
+
+        # Search for entry
+        jimaku_search = self._assemble_jimaku_search_url(video, series_name)
+        data = self._get_jimaku_response(jimaku_search['url'])
+        if not data:
+            return None
+        
+        # Determine entry
+        entry_id = None
+        entry_name = None
+        anilist_id = None
+        name_is_japanese = self._is_string_japanese(f"series_name")
+        
+        for entry in data:
+            if jimaku_search['derived_anilist_id']:
+                dict_field = 'anilist_id'
+            else:
+                entry_has_jp_name_field = 'japanese_name' in entry
+                dict_field = 'japanese_name' if (name_is_japanese and entry_has_jp_name_field) else 'english_name'
+                logger.debug(f"Attempting to get entry based on '{dict_field}' (name_is_japanese: {name_is_japanese} | entry_has_jp_name_field: {entry_has_jp_name_field})")
+            
+            # Only match first entry
+            entry_property_value = entry.get(dict_field, None)
+            if entry_property_value is not None:
+                entry_id = entry.get('id')
+                anilist_id = entry.get('anilist_id', None)
+                entry_name = entry.get('name')
+                
+                logger.info(f"Matched entry: ID: '{entry_id}', anilist_id: '{anilist_id}', name: '{entry_name}', english_name: '{entry.get('english_name')}'")
+                if entry.get("flags").get("unverified"):
+                    logger.warning(f"This entry '{entry_id}' is unverified, subtitles might be incomplete or have quality issues!")
+                
+                break
+            
+        if entry_property_value is None:
+            logger.error('Matched no entries.')
+            return None
+        
+        # Get a list of subtitles for entry
+        retry_count = 0
+        while True:
```

Comment:
I would rather refactor this to prevent while True and rely in the break:

while retry_count <= max_retries:

Reply:
Implemented in [db5f56c](https://github.com/morpheus65535/bazarr/pull/2505/commits/db5f56c2ffde69f2bde023fd79bbbdb0df575a20).